### DISCUSSION TITLE
how to use 'Seurat' using 'h5' format file?

### DISCUSSION THREAD

**Discussion by sooni9742** [satijalab/seurat#7308] [Q&A] [answered]
I am trying to analyze a single cell sequence using GSE148434 data.

I want to analyze using the 'Seurat' package, but I don't know how to do it.

I succeeded in loading the 'h5' format file. Here is my code.

```
untar("GSE148434_h5_RAW.tar")
h5.fileList <- list.files(pattern = "info.h5")
load_h5_file <- function(h5.fileList) {
  h5_data <- h5read(h5.fileList, "/")
  return(h5_data)
}
all_h5_data <- lapply(h5.fileList, load_h5_file)
```

And this is one of the data examples included in 'all_h5_data'. 'all_h5_data' is a 'Large list' format in R.

![화면 캡처 2023-05-10 190203](https://github.com/satijalab/seurat/assets/133177097/fa6bfac2-a290-4c0f-8d4a-477f11b5f7ea)

I don't know how to perform single cell rna sequencing analysis using the 'Seurat' package.

Please help me.

**Comment by Jiayi-Zheng** [comment, marked as answer]
if you file in the tar is in format of xxx.h5, do:
`expression_matrix = Read10X_h5(SINGLE_H5_FILE)
seu = CreateSeuratObject(counts = expression_matrix, min.cells = 3, min.genes = 1)`
 (and make it into your function to read in multiple files in one row)

**Reply by sooni9742** [reply]
It works! Thank you!
